packages/back-end/src/routers/github-integration/github-integration.controller.ts

```diff
@@ -0,0 +1,98 @@
+import fs from "fs";
+import { Request, NextFunction, Response } from "express";
+import { createAppAuth, createOAuthUserAuth } from "@octokit/auth-app";
+import { getOrgFromReq } from "../../services/organizations";
+import { AuthRequest } from "../../types/AuthRequest";
+import {
+  getGithubIntegrationByOrg,
+  createGithubIntegration,
+} from "../../models/GithubIntegration";
+import { createGithubUserToken } from "../../models/GithubUserTokenModel";
+import { APP_ORIGIN } from "../../util/secrets";
+
+const githubPrivateKey = fs.readFileSync(
+  process.env.GITHUB_PRIVATE_KEY_PATH || "",
```

Comment:
We'll need to store the private key file (.pem) somewhere to read from it. This was taken from the examples provided by GitHub. It's possible we could just throw the entire contents of the file into an env var, just need to be wary about variable size constraints.

Reply:
We don't have the ability to easily mount files with ECS, so we will need to store the entire contents in an env variable instead (or fetch from secrets manager etc. at runtime)